core/node/node_framework/src/implementations/layers/pk_signing_eth_client.rs

```diff
@@ -99,31 +100,31 @@ impl WiringLayer for PKSigningEthClientLayer {
             BoundEthInterfaceForBlobsResource(Box::new(signing_client_for_blobs))
         });
 
-        let signing_client_for_gateway = if let Some(client) = &input.gateway_client {
-            let private_key = self.wallets.operator.private_key();
-            let GatewayEthInterfaceResource(gateway_client) = client;
-            let chain_id = gateway_client
-                .fetch_chain_id()
-                .await
-                .map_err(WiringError::internal)?;
-            let signing_client_for_blobs = PKSigningClient::new_raw(
-                private_key.clone(),
-                input.contracts.0.chain_contracts_config.diamond_proxy_addr,
-                gas_adjuster_config.default_priority_fee_per_gas,
-                chain_id,
-                gateway_client.clone(),
-            );
-            Some(BoundEthInterfaceForL2Resource(Box::new(
-                signing_client_for_blobs,
-            )))
-        } else {
-            None
-        };
+        let signing_client_for_l2_gateway =
+            if let UniversalClient::L2(gateway_client) = input.gateway_client.0 {
```

Comment:
Nit: it's safer to use `match` for cases like this in case further variants are added to the enumeration in the future; it'd enumerate all variants explicitly.